### ISSUE TITLE
Cannot read property 'then' of undefined

### ISSUE THREAD

**Issue by Ragura** [jprichardson/node-jsonfile#117]
Getting the error `Cannot read property 'then' of undefined` and `callback is not a function` on trying to execute `jsonfile.readFile()` as a function that returns a promise. It appears its trying to call the callback version instead? If I use that version of the function, it works correctly. VSCode intellisense does list and select the version that returns a promise, so it finds the function correctly.

It fails whether the file exists or not. The path is correct.

This is my code:

```
    const jsonfile = require("jsonfile");
    const path = app.getAppPath() + "/settings.json";

    jsonfile
      .readFile(path)
      .then(obj => console.dir(obj))
      .catch(error => console.error(error));

    console.log(jsonfile);
```

**Comment by RyanZim** [comment]
What version of `jsonfile` are you using?

**Comment by Ragura** [comment]
Version 5.0.0.

**Comment by RyanZim** [comment]
Weird. Everything you're getting would point to you using an old version without promise support. Can you make a reduced test case repo?

**Comment by Ragura** [comment]
Never mind, seems it was something extremely weird on my end. I created a seperate Node project with just this package in it and it worked fine. But it still wouldn't work in mine. So I deleted the node_modules folder and ran `yarn` again. Now it works. Even though I could see zero differences in the installed jsonfile folder in node_modules (even compared the `index.js` code with the one on github).

In any case, probably not related to this package directly so I'm closing it. Sorry for the false alarm!
